pygeoapi/provider/sql.py

```diff
@@ -447,18 +448,16 @@ def _store_db_parameters(self, parameters, options):
             if not isinstance(v, dict)
         }
 
-    def _sqlalchemy_to_feature(self, item, crs_transform_out=None):
-        feature = {'type': 'Feature'}
+    def _sqlalchemy_to_feature(self, item, crs_transform_out=None,
+                               select_properties=[]):
+        feature = {'type': 'Feature', 'properties': {}}
 
-        # Add properties from item
-        item_dict = item.__dict__
-        item_dict.pop('_sa_instance_state')  # Internal SQLAlchemy metadata
-        feature['properties'] = item_dict
-        feature['id'] = item_dict.pop(self.id_field)
+        # set feature id
+        feature['id'] = item.__dict__[self.id_field]
 
         # Convert geometry to GeoJSON style
-        if feature['properties'].get(self.geom):
-            wkb_geom = feature['properties'].pop(self.geom)
+        if item.__dict__.get(self.geom) is not None:
```

Comment:
Is there a reason not to do 

`hasattr(item, self.geom)`?

Reply:
We are testing for not None, not the existence of the member.

Reply:
I mean is there a reason we are using the dictionary entity if we are accessing kv pairs that can be done on the item itself?

The dictionary representation should be the same keys as the ones used in 
hasattr and getattr, right? Why are we using the dictionary accessor at all?

Reply:
I've aliased the item to a dictionary again and accessing accordingly, as well added a docstring.